Sources/Controllers/QuickAction/ActionAddTerrainColorSchemeViewController.swift

```diff
@@ -0,0 +1,162 @@
+//
+//  ActionAddTerrainColorSchemeViewController.swift
+//  OsmAnd Maps
+//
+//  Created by Skalii on 26.09.2024.
+//  Copyright © 2024 OsmAnd. All rights reserved.
+//
+
+import Foundation
+
+@objc protocol ActionAddTerrainColorSchemeDelegate: AnyObject {
+    func onTerrainsSelected(_ items: [[String: Any]])
+}
+
+@objcMembers
+final class ActionAddTerrainColorSchemeViewController: OABaseNavbarViewController {
+
+    weak var delegate: ActionAddTerrainColorSchemeDelegate?
+
+    private var data = [(String, [TerrainMode])]()
+    private var initialPalettes: [String]
+
+    // MARK: - Initialize
+
+    init(withPalettes: [String]) {
+        self.initialPalettes = withPalettes
+        super.init(nibName: "OABaseNavbarViewController", bundle: nil)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func registerCells() {
+        addCell(OASimpleTableViewCell.reuseIdentifier)
+    }
+
+    // MARK: - UIViewController
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        tableView.allowsMultipleSelectionDuringEditing = true
+        tableView.isEditing = true
+    }
+
+    // MARK: - Base UI
+
+    override func getTitle() -> String {
+        localizedString("srtm_color_scheme")
+    }
+
+    override func getRightNavbarButtons() -> [UIBarButtonItem] {
+        [createRightNavbarButton(localizedString("shared_string_done"),
+                                 iconName: nil,
+                                 action: #selector(onRightNavbarButtonPressed),
+                                 menu: nil)]
+    }
+
+    // MARK: - Table data
+
+    override func generateData() {
+        var data = [(String, [TerrainMode])]()
+        for type in TerrainType.allCases {
+            var header = ""
+            var modes = [TerrainMode]()
+            for mode in TerrainMode.values where mode.type == type {
+                modes.append(mode)
+                if header.isEmpty {
+                    header = mode.getDescription()
+                }
+            }
+            if !modes.isEmpty {
+                modes.sort {
+                    let name0 = $0.getDefaultDescription()
```

Comment:
Don't use a variable; use $0.getDefaultDescription() directly in the relevant places and $1.getDefaultDescription() where needed